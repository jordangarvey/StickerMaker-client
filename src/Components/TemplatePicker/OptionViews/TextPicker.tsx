import React, { FC } from "react";
import { TextInput, StyleSheet } from "react-native";


/**
 * Interface for the Text Picker props
 */
interface ITextPickerProps {
	/** The currently selected text */
	currentText: string;
	/** Callback to be fired whenever the text value is changed */
	onChange(text: string): void;
}

/**
 * Component to render the Text Picker
 */
const TextPicker: FC<ITextPickerProps> = (props) => (
	<TextInput
		autoFocus={true}
		onChangeText={props.onChange}
		value={props.currentText}
	/>
);

const styles = StyleSheet.create({
});

export default TextPicker;
