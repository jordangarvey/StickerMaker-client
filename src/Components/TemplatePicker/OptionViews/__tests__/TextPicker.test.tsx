import React from "react";

import renderer from "react-test-renderer";

import TextPicker from "../TextPicker";


describe("The Text Picker componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<TextPicker
				currentText="Test"
				onChange={mockFunction}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
