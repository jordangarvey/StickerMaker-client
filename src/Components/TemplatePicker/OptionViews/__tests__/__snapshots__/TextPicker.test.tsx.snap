// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text Picker componentâ€™s snapshot renders as expected 1`] = `
<TextInput
  allowFontScaling={true}
  autoFocus={true}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
  value="Test"
/>
`;
